client/callout.go

```diff
@@ -63,13 +69,18 @@ func (c *Callout) buildRequestWithOptions(method string, url string, reqBody str
 		headers:       c.defaultHeaders,
 		timeout:       c.defaultTimeout,
 		retries:       c.defaultRetries,
+		tracer:        c.defaultTracer,
 		skipTLSVerify: c.skipTLSVerify,
 	}
 
 	for _, option := range options {
 		option(requestOpts)
 	}
 
+	if requestOpts.tracer != nil {
+		c.defaultTracer = requestOpts.tracer
+	}
+
```

Comment:
This shouldn't be necessary. The `for` loop above should take care of things.

Reply:
Hmm, my test failed without this logic. I must doing something wrong in the for-loop then.

Reply:
This is the test that will fail without the if-statement: 
```
overwrites the default setting
```